src/OrleansRuntime/Http/ExtensionMethods.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.Owin;
+using Newtonsoft.Json;
+using System;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime
+{
+    internal static class ExtensionMethods
+    {
+        public static Task ReturnJson(this IOwinContext context, object value)
```

Comment:
I think everything HTTP related should be in a separate library if possible - that's not to say that changes to the core won't be needed, just that they should be general-purpose changes, rather than HTTP-specific.